WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift

```diff
@@ -648,6 +653,49 @@ extension GutenbergViewController: GutenbergBridgeDelegate {
         mediaInserterHelper.cancelUploadOf(media: media)
     }
 
+    func gutenbergDidRequestToSetFeaturedImage(for mediaID: Int32) {
+        // dismiss current controller in order to allow the new alert controller to be presented
+        presentedViewController?.dismiss(animated: false, completion: nil)
```

Comment:
Is this to remove any view controller that might be currently presented, a sort of catch-all? If so, the only thing that I'm concerned about is that its dismissing without an animation which could be jarring to users. 

Would putting all the action sheet code inside the completion handler above work? I don't recall if it's called even if there's nothing to dismiss though, but it might be worth checking if you haven't already.

Reply:
@guarani, the bottom sheet modal (containing the image block's settings) is currently dismissed in the JS [here](https://github.com/WordPress/gutenberg/blob/53013060c4bbda4b79bff116f80a226322de47fd/packages/block-library/src/image/edit.native.js#L690) as soon as the user taps the `Set/Remove as Featured` button. But that doesn't seem to be being automatically "communicated" back to the iOS side, which is what led me adding this line of code. If I remove that line, the replace functionality fails with the following error:

`Attempt to present <UIAlertController: 0x7ff44aa16400> on <WordPress.AztecNavigationController: 0x7ff44a068200> (from <WordPress.GutenbergViewController: 0x7ff44a808e00>) which is already presenting <RCTModalHostViewController: 0x7ff437eeb340>.`

I found the open issue you created at https://github.com/WordPress/gutenberg/issues/29672 that seems to explain the reason for this behaviour but wasn't certain of the correct way to go about addressing this in my case, where the a new dialog is generated from the iOS side. Does the approach I've taken make more sense to you now with that extra context or is there something else that should be done here?

> Would putting all the action sheet code inside the completion handler above work? I don't recall if it's called even if there's nothing to dismiss though, but it might be worth checking if you haven't already.

Just to make sure I understand: Are you imagining something like this?

`presentedViewController?.dismiss(animated: false, completion: {self.showAlertForReplacingFeaturedImage(mediaID: mediaID)})
`

If so, I tested adding that at the bottom of the `gutenbergDidRequestToSetFeaturedImage` function and it works as expected. :) Would this be a better approach to take, as there will always be the bottom sheet that needs to be dismissed? 

Is there a chance that the way bottom sheets are dismissed may change in the future and negate the need for this extra dismissal in iOS (therefore break the functionality if we're relying on the completion handler)?

Reply:
> But that doesn't seem to be being automatically "communicated" back to the iOS side, which is what led me adding this line of code. If I remove that line, the replace functionality fails with the following error:

True, this sounds familiar and thanks for highlighting that existing issue. I was thinking something along the lines of this:

```
    func gutenbergDidRequestToSetFeaturedImage(for mediaID: Int32) {

        let presentAlert = { [weak self] in
            guard let `self` = self else { return }

            // ... the various guard-statements go here ...

            // ask the user to confirm changing the featured image since there's already one set
            self.showAlertForReplacingFeaturedImage(mediaID: mediaID)
        }

        if presentedViewController != nil {
            dismiss(animated: true, completion: presentAlert)
        } else {
            presentAlert()
        }
    }
```

I think this may work now and also if the bottom sheet behavior is fixed in the future, wdyt?

Reply:
Thank you so much for sharing that with me, I can see how it's a more eloquent solution and have gone ahead to implement it in 89494c318e69c07418b2f69781e2686ce2fc3e72.

Just a note: I kept `animated: false` as setting this value to `true` caused a crash and the animation is still handled on the JS side when the BottomSheet's dismissed.

Let me know if you spot any further issues here!

Reply:
Looks good! Testing now and things seem to work well 👍